neuraxle/base.py

```diff
@@ -450,10 +613,7 @@ def get_hyperparams(self, flat=True) -> HyperparameterSamples:
                 hyperparams[k] = hparams
 
         hyperparams = HyperparameterSamples(hyperparams)
-        if flat:
-            hyperparams = hyperparams.to_flat()
-        else:
-            hyperparams = hyperparams.to_nested_dict()
+
```

Comment:
Oh, I found what's wrong... I did a mistake in my previous code review. So you really need the `to_flat()` here because otherwise it's really nested for real. That's why the test isn't as I thought.